jgiven-core/src/main/java/com/tngtech/jgiven/report/asciidoc/AsciiDocReportBlockConverter.java

```diff
@@ -0,0 +1,432 @@
+        final String humanDuration = toHumanReadableDuration(duration).map(dur -> " (" + dur + ")").orElse("");
+        final String stepStatus = caseIsUnsuccessful
+            ? " " + toHumanReadableStatus(status) + humanDuration
+            : "";
+
+
+        if (extendedDescription != null && !extendedDescription.isEmpty()) {
+            return stepStatus + " +" + NEW_LINE
+                + buildIndentationFragment(depth, " ") + " _+++" + extendedDescription + "+++_";
+        } else {
+            return stepStatus;
+        }
+    }
+
+    private static String escapeTableValue(final String value) {
+        return escapeArgumentValue(value.replace("|", "\\|"));
+    }
+
+    private static String escapeArgumentValue(final String value) {
+        // TODO Is this really necessary?
+        // return "+" + value + "+";
+        return value;
+    }
+
+    private static String buildIndentationFragment(final int depth, final String symbol) {
+        return generate(() -> symbol).limit(depth + 1L).collect(joining());
+    }
+
+    private static String generateTableColSpec(final boolean withVerticalHeader, final int columnCount) {
+        return withVerticalHeader
+            ? "h," + generate(() -> "1").limit(columnCount - 1L).collect(joining(","))
+            : generate(() -> "1").limit(columnCount).collect(joining(","));
+    }
+
+    private static String toAsciiDocTag(final ExecutionStatus executionStatus) {
+        switch (executionStatus) {
+            case SCENARIO_PENDING:
+                // fall through
+            case SOME_STEPS_PENDING:
+                return "scenario-pending";
+            case SUCCESS:
+                return "scenario-successful";
+            case FAILED:
+                return "scenario-failed";
+            default:
+                return "scenario-unknown";
+        }
+    }
+
+    private static String toHumanReadableStatus(final ExecutionStatus executionStatus) {
+        switch (executionStatus) {
+            case SCENARIO_PENDING:
+                // fall through
+            case SOME_STEPS_PENDING:
+                return ICON_BANNED;
+            case SUCCESS:
+                return ICON_CHECK_MARK;
+            case FAILED:
+                return ICON_EXCLAMATION_MARK;
+            default:
+                return executionStatus.toString();
+        }
+    }
+
+    private static String toHumanReadableStatus(final StepStatus stepStatus) {
+        switch (stepStatus) {
+            case PASSED:
+                return ICON_CHECK_MARK;
+            case FAILED:
+                return ICON_EXCLAMATION_MARK;
+            case SKIPPED:
+                return ICON_STEP_FORWARD;
+            case PENDING:
+                return ICON_BANNED;
+            default:
+                return stepStatus.toString();
+        }
+    }
+
+    private static Optional<String> toHumanReadableDuration(final long nanos) {
```

Comment:
I take issue with this method, because it doesn't exactly announce what it does. (and one may also argue that it does two things), because the method name, in conjunction with the return type suggests that the method will attempt a conversion, which may fail, when in fact it makes a decision whether it should convert AND converts the input.

Unfortunately, I don't have a nifty alternative at hand right now :(